### PR TITLE
updated the git signatures to the user who is logged in instead of th…

### PR DESCRIPTION
**Pull request by dragonprevost**
This pull request has a new function to generate Signatures based on the account that has logged in instead of the Oauth application account. Everywhere that `repo.defaultSignature()` is called, has been replaced with `getSignature()`.

### CODE REVIEW

File: app/misc/authenticate.ts [hunk 1]
```diff
@@ -535,3 +535,14 @@ function getUsername(){
   // Return the users username
   return account.login;
 }
+
+// Get a git signature for the account that is logged in.
+function getSignature(){
+  // Set email to the email in account object
+  var email = account.email;
```
Comment (colinwerner) [inline app/misc/authenticate.ts]: Probably need to check for account (in the case when I am not logged in)?

image

Comment (dragonprevost) [reply app/misc/authenticate.ts]: Good catch, did this happen when you differed logging in?

Comment (colinwerner) [reply app/misc/authenticate.ts]: Yes. "continue without sign in" then stash some changes.

Comment (dragonprevost) [reply app/misc/authenticate.ts]: I updated the get signature function to take the repository as a parameter and return the default signature if the user has not logged in. Give it a shot and let me know if it solves this issue! :)

Comment (jerushachua) [reply app/misc/authenticate.ts]: Is '@users.noreply.github.com' the only type of email that will appear?

Comment (dragonprevost) [reply app/misc/authenticate.ts]: If the account has a public email address, then that public email address will be applied to the signature, if not the email will follow the format of `<uesrname>@users.noreply.github.com`.

**Pull request merged**